From 551125c33e06b7aa7b4cdfafbc800ceda6c2237a Mon Sep 17 00:00:00 2001
From: Sidraya <sidraya.bj@pathpartnertech.com>
Date: Tue, 25 Apr 2023 17:14:56 -0500
Subject: [PATCH 399/508] media: platform: Makefile: Fix build failure in
 parallel build Env.

Fix build failure in parallel build Env.
prevent building source code under common directory
twice which leads to build fail.

Signed-off-by: Sidraya <sidraya.bj@pathpartnertech.com>
Signed-off-by: Darren Etheridge <detheridge@ti.com>
---
 .../platform/img/vxe-vxd/{decoder => }/Makefile   | 138 +++++++++++-------
 .../media/platform/img/vxe-vxd/encoder/Makefile   |  56 -------
 2 files changed, 86 insertions(+), 108 deletions(-)
 rename drivers/media/platform/img/vxe-vxd/{decoder => }/Makefile (50%)
 delete mode 100644 drivers/media/platform/img/vxe-vxd/encoder/Makefile

diff --git a/drivers/media/platform/img/vxe-vxd/decoder/Makefile b/drivers/media/platform/img/vxe-vxd/Makefile
index ac0d4e830802..8f159aa0f56a 100644
--- a/drivers/media/platform/img/vxe-vxd/Makefile
+++ b/drivers/media/platform/img/vxe-vxd/Makefile
@@ -46,67 +46,55 @@ ERROR_RECOVERY_SIMULATION ?=n
 # If unsure, say Y
 CAPTURE_CONTIG_ALLOC ?=y
 
-#VXD
-vxd-dec-y += vxd_core.o
-
-#PVDEC
-vxd-dec-y += vxd_pvdec.o
-
-#MEM_MGR
-vxd-dec-y += ../common/img_mem_man.o ../common/img_mem_unified.o
-vxd-dec-y += ../common/imgmmu.o
-
-#Utilities
-vxd-dec-y += ../common/lst.o ../common/dq.o
-vxd-dec-y += ../common/resource.o
-vxd-dec-y += dec_resources.o
-vxd-dec-y += ../common/rman_api.o
-vxd-dec-y += pixel_api.o
-vxd-dec-y += vdecdd_utils_buf.o
-vxd-dec-y += vdecdd_utils.o
-
-#MMU
-vxd-dec-y += ../common/talmmu_api.o
-vxd-dec-y += ../common/pool.o
-vxd-dec-y += ../common/hash.o
-vxd-dec-y += ../common/ra.o
-vxd-dec-y += ../common/addr_alloc.o
-vxd-dec-y += ../common/work_queue.o
-vxd-dec-y += vdec_mmu_wrapper.o
-
-#DECODER
-vxd-dec-y += ../common/pool_api.o ../common/idgen_api.o
-vxd-dec-y += hw_control.o
-vxd-dec-y += vxd_int.o
-vxd-dec-y += translation_api.o
-vxd-dec-y += decoder.o
-vxd-dec-y += core.o
-
-#BSPP
-vxd-dec-y += swsr.o
-vxd-dec-y += h264_secure_parser.o
-vxd-dec-y += bspp.o
-
-#UM INTERFACE & SYSDEV
-vxd-dec-y += vxd_dec.o
-
-vxd-dec-y += vxd_v4l2.o
+vxd-dec-y += common/img_mem_man.o \
+	common/img_mem_unified.o \
+	common/imgmmu.o \
+	common/pool_api.o \
+	common/idgen_api.o \
+	common/talmmu_api.o \
+	common/pool.o \
+	common/hash.o \
+	common/ra.o \
+	common/addr_alloc.o \
+	common/work_queue.o \
+	common/lst.o \
+	common/dq.o \
+	common/resource.o \
+	common/rman_api.o \
+
+vxd-dec-y += decoder/vxd_core.o \
+	decoder/vxd_pvdec.o \
+	decoder/dec_resources.o \
+	decoder/pixel_api.o \
+	decoder/vdecdd_utils_buf.o \
+	decoder/vdecdd_utils.o \
+	decoder/vdec_mmu_wrapper.o \
+	decoder/hw_control.o \
+	decoder/vxd_int.o \
+	decoder/translation_api.o \
+	decoder/decoder.o \
+	decoder/core.o \
+	decoder/swsr.o \
+	decoder/h264_secure_parser.o \
+	decoder/bspp.o \
+	decoder/vxd_dec.o \
+	decoder/vxd_v4l2.o \
 
-ifeq ($(DEBUG_DECODER_DRIVER), y)
-ccflags-y   += -DDEBUG_DECODER_DRIVER
-ccflags-y   += -DDEBUG
-endif
 
 ifeq ($(HAS_HEVC),y)
 ccflags-y   += -DHAS_HEVC
-vxd-dec-y += hevc_secure_parser.o
+vxd-dec-y += decoder/hevc_secure_parser.o
 endif
 
 ifeq ($(HAS_JPEG),y)
 ccflags-y   += -DHAS_JPEG
-vxd-dec-y += jpeg_secure_parser.o
+vxd-dec-y += decoder/jpeg_secure_parser.o
 endif
 
+ifeq ($(DEBUG_DECODER_DRIVER), y)
+ccflags-y   += -DDEBUG_DECODER_DRIVER
+ccflags-y   += -DDEBUG
+endif
 
 ifeq ($(ERROR_CONCEALMENT),y)
 ccflags-y   += -DERROR_CONCEALMENT
@@ -126,4 +114,50 @@ endif
 
 obj-$(CONFIG_VIDEO_IMG_VXD_DEC) += vxd-dec.o
 
-ccflags-y += -I$(srctree)/drivers/media/platform/img/vxe-vxd/common
+# (1)
+# This config, if enabled, enables all the debug traces in
+# encoder driver. Enable it only for debug purpose
+# Keep it always disabled for release codebase
+DEBUG_ENCODER_DRIVER ?=n
+
+# (3)
+# This config enables encoder performance profiling
+# keep it always disabled. Enable it only for profiling in development
+# environments.
+ENABLE_PROFILING ?=n
+
+vxe-enc-y += common/img_mem_man.o \
+	common/img_mem_unified.o \
+	common/talmmu_api.o \
+	common/addr_alloc.o \
+	common/lst.o \
+	common/hash.o \
+	common/ra.o \
+	common/pool.o \
+	common/rman_api.o \
+	common/dq.o \
+	common/idgen_api.o \
+	common/imgmmu.o \
+	common/work_queue.o \
+
+vxe-enc-y += encoder/vxe_v4l2.o \
+	encoder/vxe_enc.o \
+	encoder/topaz_device.o \
+	encoder/topazmmu.o \
+	encoder/topaz_api.o \
+	encoder/topaz_api_utils.o \
+	encoder/header_gen.o \
+	encoder/mtx_fwif.o \
+
+obj-$(CONFIG_VIDEO_IMG_VXE_ENC) += vxe-enc.o
+
+ifeq ($(DEBUG_ENCODER_DRIVER), y)
+ccflags-y   += -DDEBUG_ENCODER_DRIVER
+ccflags-y   += -DDEBUG
+endif
+
+ifeq ($(ENABLE_PROFILING),y)
+ccflags-y	+= -DENABLE_PROFILING
+endif
+
+ccflags-y += -I$(srctree)/drivers/media/platform/img/vxe-vxd/common/
